File: src/ground-ui-service/src/types/mission.ts
export interface Mission {
  id: string;
  name: string;
  description?: string | null;
  isActive: boolean;
  createdAt: string;
}

export interface MissionCreateRequest {
  name: string;
  description?: string | null;
}

export interface MissionUpdateRequest {
  name: string;
  description?: string | null;
  isActive: boolean;
}

File: src/satellite-autonomy-service/Sentinel.Core/Entities/Mission.cs
namespace Sentinel.Core.Entities;

public sealed class Mission
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public DateTimeOffset CreatedAt { get; set; }

    public bool IsActive { get; set; }

    public ICollection<Satellite> Satellites { get; set; } = new List<Satellite>();

    public static Mission Create(string name, string? description)
    {
        return new Mission
        {
            Id = Guid.NewGuid(),
            Name = name.Trim(),
            Description = string.IsNullOrWhiteSpace(description) ? null : description.Trim(),
            IsActive = true,
            CreatedAt = DateTimeOffset.UtcNow
        };
    }

    public void Update(string name, string? description, bool isActive)
    {
        Name = name.Trim();
        Description = string.IsNullOrWhiteSpace(description) ? null : description.Trim();
        IsActive = isActive;
    }
}
